docs/src/example.md

```diff
@@ -31,13 +31,15 @@ There are only two small technical things we need to explain at this point befor
 First of all our [`AbstractManifold`](@ref)`{𝔽}` has a parameter `𝔽`.
 This parameter indicates the [`number_system`](@ref) the manifold is based on, for example `ℝ` for real manifolds, which is short for [`RealNumbers`](@ref)`()` or `ℂ` for complex manifolds, a shorthand for [`ComplexNumbers`](@ref)`()`.
 
-Second, this interface usually provides both an allocating and a mutating variant of each function, for example for the [`exp`](@ref)onential map [implemented below](@ref manifold-tutorial-fn) this interface provides `exp(M,p,X)` to compute the exponential map and `exp!(M, q, p, X)` to compute the exponential map in the memory provided by `q`, mutating that input.
+Second, this interface usually provides both an allocating and a mutating variant of each function, for example for the [`exp`](@ref)onential map [implemented below](@ref manifold-tutorial-fn) this interface provides `exp(M, p, X)` to compute the exponential map and `exp!(M, q, p, X)` to compute the exponential map in the memory provided by `q`, mutating that input.
 the convention is, that the manifold is the first argument -- in both function variants -- the mutating variant then has the input to be mutated in second place, and the remaining parameters are again the same (`p`and `X` here).
 We usually refer to these two variants of the same function as the allocating (`exp`) function and the mutating (`exp!`) one.
```

Comment:
since we are at it, I think in Manopt I have switched to calling the `!` variants the in-place variants (instead of mutating),

Reply:
Sure, I will rename it.

Reply:
I've renamed `mutating` to `in-place`, but for now I've left `non-mutating`.

Reply:
that's fine with me. I even called the “non-mutating” one “allocating”, but I think that is fine giving it a negative (non-) name here.